api/v1/db/oracledb/game-dao.js

```diff
@@ -0,0 +1,96 @@
+
+const sqlQuery = `
+    SELECT CN.CARD_NUMBER, CS.SUIT, G.GAME_ID, R.ROUND, G.MAXIMUM_BET,
+    G.MINIMUM_BET, G.BET_POOL
+    FROM GAMES G
+    INNER JOIN ROUNDS R ON G.ROUND_ID = R.ROUND_ID
+    LEFT OUTER JOIN TABLE_CARDS TC ON TC.GAME_ID = G.GAME_ID
+    LEFT OUTER JOIN CARDS C ON TC.CARD_ID = C.CARD_ID
+    LEFT OUTER JOIN CARD_NUMBERS CN ON C.CARD_NUMBER_ID = CN.CARD_NUMBER_ID
+    LEFT OUTER JOIN CARD_SUITS CS ON C.CARD_SUIT_ID = CS.SUIT_ID
+    `;
+/**
+ * @summary Return a list of games
+ * @function
+ * @param {Object} query query object that contains useful information to process.
+ * @returns {Promise<Object[]>} Promise object represents a list of games
+ */
+const getGames = async (query) => {
+  const connection = await conn.getConnection();
+  const round = query ? query.round : null;
+
+  try {
+    const sqlParams = {};
+    if (round) {
+      sqlParams.round = round;
+    }
+    const getGamesSqlQuery = `${sqlQuery} ${round ? 'WHERE R.ROUND = :round' : ''}`;
+    const rawGamesResponse = await connection.execute(getGamesSqlQuery, sqlParams);
+    const rawGames = rawGamesResponse.rows;
+    const serializedGames = serializeGames(rawGames, query);
+    return serializedGames;
+  } finally {
+    connection.close();
+  }
+};
+
+const getGameById = async (id) => {
+  const connection = await conn.getConnection();
+  try {
+    const sqlParams = [id];
+    const getGameByIdSqlQuery = `${sqlQuery} ${id ? 'WHERE G.GAME_ID = :id' : ''}`;
+    const rawGamesResponse = await connection.execute(getGameByIdSqlQuery, sqlParams);
+    const rawGames = rawGamesResponse.rows;
+    if (_.isEmpty(rawGames)) {
+      return undefined;
+    }
+    if (_.keys(_.groupBy(rawGames, 'GAME_ID')).length > 1) {
+      throw new Error('Expect a single object but got multiple results.');
+    } else {
+      const serializedGame = serializeGame(rawGames);
+      return serializedGame;
+    }
+  } finally {
+    connection.close();
+  }
+};
+
+const postGame = async (body) => {
+  const connection = await conn.getConnection();
+  body = body.data.attributes;
+  body.outId = {
+    type: oracledb.NUMBER,
+    dir: oracledb.BIND_OUT,
+  };
+  body.round = body.round.charAt(0).toUpperCase();
+
+  const { memberIds } = body;
+
+  delete body.memberIds;
+  const postSqlQuery = `INSERT INTO GAMES (ROUND_ID, MINIMUM_BET, MAXIMUM_BET, BET_POOL) VALUES
+  (:round, :minimumBet, :maximumBet, :betPool) RETURNING GAME_ID INTO :outId`;
+  const rawGames = await connection.execute(postSqlQuery, body, { autoCommit: true });
+
+  const gameId = rawGames.outBinds.outId[0];
+  _.forEach(memberIds, async (id) => {
+    const playerSqlParams = {
+      gameId,
+      memberId: id,
+    };
+    const playerSqlQuery = `INSERT INTO PLAYERS (MEMBER_ID, GAME_ID, PLAYER_BET, STATUS_ID) VALUES
+    (:memberId, :gameId, 0, 'CH')`;
+    await connection.execute(playerSqlQuery, playerSqlParams, { autoCommit: true });
+  });
```

Comment:
Using `_.forEach()` with asynchronous functions will cause all of these queries to be fired off and execution flow will immediately continue after the `_.forEach()`, which isn't what you want. One solution you may want to take a look at is [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)